kodilla-good-patterns/src/main/java/com/kodilla/goodPatterns/challenges/selling/ItemSellService.java

```diff
@@ -1,12 +1,18 @@
 package com.kodilla.goodPatterns.challenges.selling;
 
+import com.kodilla.goodPatterns.challenges.Food2Door.ProducerProduct;
+import com.kodilla.goodPatterns.challenges.Food2Door.ProductAvailable;
 import com.kodilla.goodPatterns.challenges.products.Invoice;
+import com.kodilla.goodPatterns.challenges.products.Item;
+import com.kodilla.goodPatterns.challenges.users.Producer;
 import com.kodilla.goodPatterns.challenges.users.User;
 
+import java.util.List;
+
 public class ItemSellService implements SellService {
 
-    public boolean sell(User user, Invoice invoice) {
-        if (user != null && invoice != null) {
+    public boolean sell(User user, Invoice invoice, Producer producer, ProducerProduct producerProduct) {
+        if (user != null || invoice != null || invoice.getQuantity() >= producerProduct.getQuantity()) {
```

Comment:
Odpowiadajac na twoje pytanie, mozna to zrobic na wiele sposobow. Zgodnie z DDD (Domain Driven Design), powinienes przesylac kolejne produkty zawarte w invoice (pod itemami) do producenta (Producer) oraz odpytywac czy dany produkt jest dostepny. 
Czyli producer, moze miec metode ktora przyjmuje product i ilosc na wejsciu i zwraca booleana na wyjsciu czy dane zamowienie moze zostac zrealizowane